lib/driver.js

```diff
@@ -467,12 +467,12 @@ export class XCUITestDriver extends BaseDriver {
       // ensure WDA gets our defaults instead of whatever its own might be
       await this.updateSettings(wdaSettings);
 
-      // turn on mjpeg stream reading if requested
-      if (this.opts.mjpegScreenshotUrl) {
-        this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
-        this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
-        await this.mjpegStream.start();
+      if (this.isRealDevice() && this.opts.mjpegServerPort) {
```

Comment:
I've checked the driver code one more time and it looks like that is a simulator is used then we should simple set `usePortForwarding` option to `false` in the corresponding `DEVICE_CONNECTIONS_FACTORY.requestConnection` call

Reply:
Agree! For simulator we don't need to port forward, this will only execute for real device right. So why worry?

Reply:
Sorry, I don't get understand this. I will take this offline for faster communication.